### PR TITLE
[#2158] Add More Documentation for Title Component

### LINKED ISSUE #2158: Add More Documentation for Title Component
### What feature(s) would you like to see in RepoSense

- Add more documentation for the new Title component e.g. example code, further reading, behaviour for edge cases.
- Add a Title component for the preview dashboard.

### Additional context

#2102

### PR DESCRIPTION
**Pull request by sopa301**
Fixes #2158.

### Proposed commit message

```
Add More Documentation for Title Component

The documentation for the title component is sparse. Adding more
details on how to use it will make it easier for users to add their own
titles to their reports.

Let's update the documentation to include a section for the title
component, and add the title component to the default view.
```

### Other information

Since the preview deployment uses the default build of the RepoSense report, all future builds will use the title component by default.

### CODE REVIEW

**Review by asdfghjkxd — APPROVED** [review]
LGTM!

File: docs/ug/customizingReports.md [hunk 1]
```diff
@@ -59,3 +59,17 @@ In both instances, it is **necessary to commit any changes** for them to be dete
...
 3\. Add a git `.mailmap` file at the top-level of the repository, specifying mapped authors/commiters and/or e-mail addresses as per [gitmailmap documentation](https://git-scm.com/docs/gitmailmap). Any mappings specified here will be applied by git before all other RepoSense configurations. Configuration via `.mailmap` is particularly useful if you want the mapping to apply for all git commands as well instead of just for RepoSense.
+
+<!-- ------------------------------------------------------------------------------------------------------ -->
+
+### Add a Title
+
+A title component can be added by creating a file titled `title.md` in the assets directory. You can specify the assets directory according to the reference below:
+  {{ embed("Appendix: **CLI syntax reference**", "cli.md") }}
+
+The title can render a combination of HTML and plaintext ([example](title.md)), and will appear on the top of the left panel as shown below:
```
Comment (ckcherry23) [inline docs/ug/customizingReports.md:70]: The example title file here opens up within the user guide documentation itself as another page. Instead, it should appear as code, perhaps pointing to the code file in GitHub. Can you change the link here accordingly?

File: docs/ug/customizingReports.md [hunk 2]
```diff
@@ -59,3 +59,17 @@ In both instances, it is **necessary to commit any changes** for them to be dete
...
+![Title Component Example](../images/title-example.png)
+
+Do note that the width of the title is bound by the width of the left panel.
+
+For more information on how to create a Markdown file, see the [Markdown Guide](https://www.markdownguide.org/).
```
Comment (ckcherry23) [inline docs/ug/customizingReports.md:75]:
```suggestion
For more information on how to use Markdown, see the [Markdown Guide](https://www.markdownguide.org/).
```

File: docs/ug/customizingReports.md (hunk 1 above)
Comment (ckcherry23) [inline docs/ug/customizingReports.md:70]: We can indicate that the title uses Markdown syntax explicitly.

```suggestion
The title can render a combination of Markdown/HTML and plaintext ([example](title.md)), and will appear on the top of the left panel as shown below:
```

**Review by ckcherry23 — CHANGES_REQUESTED** [review]

File: docs/ug/customizingReports.md [hunk 3]
```diff
@@ -59,3 +59,17 @@ In both instances, it is **necessary to commit any changes** for them to be dete
 </box>
 
 3\. Add a git `.mailmap` file at the top-level of the repository, specifying mapped authors/commiters and/or e-mail addresses as per [gitmailmap documentation](https://git-scm.com/docs/gitmailmap). Any mappings specified here will be applied by git before all other RepoSense configurations. Configuration via `.mailmap` is particularly useful if you want the mapping to apply for all git commands as well instead of just for RepoSense.
+
+<!-- ------------------------------------------------------------------------------------------------------ -->
+
+### Add a Title
```
Comment (ckcherry23) [inline docs/ug/customizingReports.md:65]:
```suggestion
### Add a title
```

**Review by ckcherry23 — APPROVED** [review]
LGTM!

**Review by vvidday — APPROVED** [review]
LGTM

**Comment by damithc** [comment]
> Since the preview deployment uses the default build of the RepoSense report, all future builds will use the title component by default.

@sopa301 Does this mean all dashboard will have a default title.md? That shouldn't be the case, as that will force users to override the title.md, turning this opt-in feature into an opt-out feature.

**Comment by sopa301** [comment]
Yes prof, I'll create a PR to remove it. If we still want the preview dashboard to have the title, I think we might need to create a separate config for the preview dashboard. 

Another feature that will also stand to gain from this arrangement would be the blurbs.

**Pull request merged**